aiohttp/http_writer.py

```diff
@@ -71,10 +70,9 @@ def _write(self, chunk: bytes) -> None:
         size = len(chunk)
         self.buffer_size += size
         self.output_size += size
-
-        if self._transport is None or self._transport.is_closing():
+        if not self._protocol.connected:
             raise ConnectionResetError("Cannot write to closing transport")
-        self._transport.write(chunk)
+        self.transport.write(chunk)
```

Comment:
You could do something funky with a `TypeGuard`, but you'd probably need to make the protocol generic in order to make it work. Probably easier to just add an assert here.

```suggestion
        assert self.transport is not None  # Must be true if connected.
        self.transport.write(chunk)
```

Reply:
Saying that, there is actually a test failing because this is None. Not sure how that happens..

Reply:
Looks like a mocking issue in the test. I'll revert the None check to make mypy happy.